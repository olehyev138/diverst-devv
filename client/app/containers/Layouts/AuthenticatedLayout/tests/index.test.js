import React from 'react';
import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';

import { StyledAuthenticatedLayout } from '../index';
const AuthenticatedLayoutNaked = unwrap(StyledAuthenticatedLayout);

/**
 * TODO:
 *  - test it accepts & renders a child component
 *  - test it redirects if not authenticated
 *  - test it uses props correctly
 */


const props = {
  renderAppBar: true,
  drawerOpen: true,
  drawerToggleCallback: jest.fn(),
  position: '',
  isAdmin: true,
  component: jest.fn()
};

describe('<AuthenticatedLayout />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<AuthenticatedLayoutNaked {...props} />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<AuthenticatedLayoutNaked {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
