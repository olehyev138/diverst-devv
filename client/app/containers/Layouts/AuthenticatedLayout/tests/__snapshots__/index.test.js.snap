// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticatedLayout /> Should render and match the snapshot 1`] = `<NotAuthenticated />`;
